whisper.py

```diff
@@ -872,23 +872,31 @@ def info(path):
   return None
 
 
-def fetch(path, fromTime, untilTime=None, now=None):
-  """fetch(path,fromTime,untilTime=None)
+def fetch(path, fromTime, untilTime=None, now=None, archiveToSelect=None):
+  """fetch(path,fromTime,untilTime=None,archiveToSelect=None)
 
 path is a string
 fromTime is an epoch time
 untilTime is also an epoch time, but defaults to now.
+archiveToSelect is the requested granularity, but defaults to None.
 
 Returns a tuple of (timeInfo, valueList)
 where timeInfo is itself a tuple of (fromTime, untilTime, step)
 
 Returns None if no data can be returned
 """
+
+  #Parse granularity
+  if archiveToSelect:
+    retentionStr = str(archiveToSelect)+":1s"
```

Comment:
This can be slightly more efficient if you specify ":1", I'd also move this down into `file_fetch` so that the `archiveToSelect` parameter has the same semantics in both functions.